back/src/test/kotlin/com/sos/smartopenspace/controllers/TalkControllerTest.kt

```diff
@@ -0,0 +1,89 @@
+package com.sos.smartopenspace.controllers
+
+import com.sos.smartopenspace.domain.*
+import com.sos.smartopenspace.persistence.OpenSpaceRepository
+import com.sos.smartopenspace.persistence.RoomRepository
+import com.sos.smartopenspace.persistence.TalkRepository
+import com.sos.smartopenspace.persistence.UserRepository
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.test.context.ActiveProfiles
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers
+import org.springframework.transaction.annotation.Transactional
+import java.time.LocalDate
+import java.time.LocalTime
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@ActiveProfiles("test")
+@Transactional
+class TalkControllerTest {
+
+    @Autowired
+    lateinit var mockMvc: MockMvc
+
+
+    @Autowired
+    lateinit var userRepository: UserRepository
+    @Autowired
+    lateinit var openSpaceRepository: OpenSpaceRepository
+    @Autowired
+    lateinit var talkRepository: TalkRepository
+    @Autowired
+    lateinit var roomRepository: RoomRepository
+
+    @Test
+    fun `schedule a talk returns an ok status response`() {
+        val organizer = anySavedUser()
+        val talk = anySavedTalk()
+        val room = anySavedRoom()
+        openSpaceRepository.save(anyOpenSpaceWith(talk, organizer, room))
+        val time = LocalTime.parse("09:30")
+
+        mockMvc.perform(
+                MockMvcRequestBuilders.put("/talk/schedule/${organizer.id}/${talk.id}/${room.id}/${time}")
+        ).andExpect(MockMvcResultMatchers.status().isOk)
+
+    }
+
+    @Test
+    fun `when cannot be schedule a talk returns a bad request response`() {
```

Comment:
```suggestion
    fun `when a talk cannot be scheduled it should return a bad request response`() {
```

Ahí vi que los tests unitarios se encargan de las clases de equivalencia que hacen que una charla no pueda ser agendada.